libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -389,7 +389,9 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 	if (_node.isConstructor())
 	{
 		if (_node.annotation().contract)
-			visibility = _node.annotation().contract->abstract() ? Visibility::Internal : Visibility::Public;
+			visibility =
```

Comment:
This bug can also be solved by changing the import instead of the export. The downside of changing the export is that it might need a change in tooling. Is changing the import difficult?

Reply:
So if a contract specifies public for the constructor, and it's AST gets imported. Should it still have the warning? I think it should. If we want this behaviour, we need to change it at export.

Reply:
There are many other warnings we don't retain during import, but if you want to keep it then I'm fine with keeping it. Please put it as a breaking change to the AST into the changelog, though.

Reply:
I'll change changelog entry to AST Change. Can this go to develop? Or does it really have to go to breaking.

I don't have a strong opinion on the warning during import. But I would prefer if this didn't go into breaking.